packages/adapter/addon/index.ts

```diff
@@ -9,6 +9,8 @@ type ShimModelClass = import('@ember-data/store/-private/system/model/shim-model
 type Store = import('@ember-data/store/-private/system/core-store').default;
 type Snapshot = import('ember-data/-private').Snapshot;
 type SnapshotRecordArray = import('@ember-data/store/-private/system/snapshot-record-array').default;
+// eslint-disable-next-line no-restricted-globals
+type AdapterPromise<T> = Promise<T> | RSVPPromise<T>;
```

Comment:
Why can't it be Promise<T>?  Do we sometimes rely on RSVP internals?

Reply:
We sometimes return an RSVPPRomise and we sometimes return a Native Promise, this depends entirely on whether we've created the promise or whether we've received it from a native API.

This mixing is why I think we should do an audit at some point to make sure we haven't accidentally used a native promise somewhere, I'm fairly sure we haven't, but we do have to expect that a promise received/returned could be either.

fwiw `yarn problems` sees no issue with the return types having been just Promise.

typescript-eslint was complaining for things that were typed Promise as an argument or as a return, not because of a ts compiler error but due to the `no-restricted-globals` lint rule. Since we mix the returns I figured I'd go ahead and set the AdapterPromise to both and only disable the rule once vs for the entire file or at every usage of Promise as type. We can probably do this single-location lint override and only mirror Promise as AdapterPromise vs both types.

The best approach is a good question for @mike-north or @chriskrycho.